graveler/graveler.go

```diff
@@ -841,6 +841,54 @@ func (g *graveler) Commit(ctx context.Context, repositoryID RepositoryID, branch
 	return newCommit, nil
 }
 
+var (
+	ErrDirtyBranch  = errors.New("can't apply tree on dirty branch")
+	ErrTreeNotFound = errors.New("tree not found")
+)
+
+func (g *graveler) CommitExistingTree(ctx context.Context, repositoryID RepositoryID, branchID BranchID, treeID TreeID, committer string, message string, metadata Metadata) (CommitID, error) {
+	cancel, err := g.branchLocker.AquireMetadataUpdate(repositoryID, branchID)
+	if err != nil {
+		return "", fmt.Errorf("acquire metadata update: %w", err)
+	}
+	defer cancel()
+	repo, err := g.RefManager.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return "", fmt.Errorf("get repository: %w", err)
```

Comment:
Consider
```suggestion
		return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
```
otherwise the error is much less actionable.  (Similarly below and throughout)

Reply:
Done.